File: src/test/java/com/flexpoker/game/command/aggregate/tablebalancer/TableBalancerTestUtils.java
package com.flexpoker.game.command.aggregate.tablebalancer;

import java.util.HashMap;
import java.util.Map;
import java.util.Set;
import java.util.UUID;
import java.util.stream.Collectors;
import java.util.stream.Stream;

public class TableBalancerTestUtils {

    /**
     * Helper method to create a set of table to player maps. The subject
     * table/number of players is treated specially in order to keep some
     * control over the tests.
     * 
     * @param subjectTableId
     * @param numberOfPlayersAtEachTable
     * @return
     */
    public static Map<UUID, Set<UUID>> createTableToPlayersMap(
            UUID subjectTableId, int numberOfSubjectTablePlayers,
            int... numberOfPlayersAtOtherTables) {
        var tableToPlayersMap = new HashMap<UUID, Set<UUID>>();
        tableToPlayersMap.put(subjectTableId, createRandomSetOfPlayerIds(numberOfSubjectTablePlayers));

        for (var i = 0; i < numberOfPlayersAtOtherTables.length; i++) {
            tableToPlayersMap.put(UUID.randomUUID(), createRandomSetOfPlayerIds(numberOfPlayersAtOtherTables[i]));
        }

        return tableToPlayersMap;
    }

    private static Set<UUID> createRandomSetOfPlayerIds(int numberOfPlayers) {
        return Stream.iterate(0, e -> e + 1).limit(numberOfPlayers)
                .map(x -> UUID.randomUUID()).collect(Collectors.toSet());
    }

    public static Map<UUID, Integer> createDefaultChipMapForSubjectTable(
            UUID subjectTableId, Map<UUID, Set<UUID>> tableToPlayersMap) {
        return tableToPlayersMap.get(subjectTableId).stream()
                .collect(Collectors.toMap(x -> x, x -> 100));
    }

}

File: src/test/kotlin/com/flexpoker/game/command/aggregate/tablebalancer/TableBalancerTestUtils.kt
package com.flexpoker.game.command.aggregate.tablebalancer

import com.flexpoker.util.toPMap
import com.flexpoker.util.toPSet
import org.pcollections.PMap
import org.pcollections.PSet
import java.util.UUID

object TableBalancerTestUtils {

    /**
     * Helper method to create a set of table to player maps. The subject
     * table/number of players is treated specially in order to keep some
     * control over the tests.
     */
    fun createTableToPlayersMap(subjectTableId: UUID, numberOfSubjectTablePlayers: Int,
                                vararg numberOfPlayersAtOtherTables: Int): PMap<UUID, PSet<UUID>> {
        val tableToPlayersMap = HashMap<UUID, PSet<UUID>>()
        tableToPlayersMap[subjectTableId] = createRandomSetOfPlayerIds(numberOfSubjectTablePlayers).toPSet()
        for (element in numberOfPlayersAtOtherTables) {
            tableToPlayersMap[UUID.randomUUID()] = createRandomSetOfPlayerIds(element).toPSet()
        }
        return tableToPlayersMap.toPMap()
    }

    private fun createRandomSetOfPlayerIds(numberOfPlayers: Int): Set<UUID> {
        return (0 until numberOfPlayers).map { UUID.randomUUID() }.toSet()
    }

    fun createDefaultChipMapForSubjectTable(subjectTableId: UUID, tableToPlayersMap: Map<UUID, Set<UUID>>): Map<UUID, Int> {
        return tableToPlayersMap[subjectTableId]!!.associateWith { 100 }
    }

}